### ISSUE TITLE
npx react-native info and npx react-native init fails with error

### ISSUE THREAD

**Issue by rjatin007** [react-native-community/cli#1681]
## Environment
Output of npx react-native info :- 
error Failed to load configuration of your project.

## Description

npx react-native fails with error : - Failed to load configuration of your project.
I have run this with --verbose too
<img width="1676" alt="Screenshot 2022-08-18 at 7 46 45 PM" src="https://user-images.githubusercontent.com/30732342/185417557-8c2e8af0-4f5f-4f27-9528-9ac3a4cbbdf5.png">

## Reproducible Demo

npx react-native info
<img width="1666" alt="Screenshot 2022-08-18 at 7 45 10 PM" src="https://user-images.githubusercontent.com/30732342/185417219-6aaa9d08-3130-4d41-a61f-56fa2e01ee28.png">

**Comment by nazrdogan** [comment]
same issue

**Comment by olivierlesnicki** [comment]
I have the same issue: https://github.com/react-native-community/cli/issues/1681

**Comment by xiaohui-gsv** [comment]
I have the issue with `npx react-native init`, but the error is different, I deleted the `.npm/_npx` folder, and it works.
You can refer to [1692](https://github.com/react-native-community/cli/issues/1692) for more details to see if it can help.

**Comment by rjatin007** [comment]
@xiaohui-gsv  mine doesn't even start the command

**Comment by 0xf2dd55** [comment]
Hi, I have the same issue recently.
I did create one new project successfully and I still can run it but I can't create a new one or try to run another project I downloaded from a repo.
I do not have a cli installed globally and only have a few npm packages installed globally :
├── create-react-app@5.0.1
├── npm-check-updates@15.0.4
├── npm@8.18.0
├── patch-package@6.4.7
├── solc@0.8.15
└── truffle@5.5.20

Here are the npx react-native outputs :

`steph@StephMBA BlueWallet % npx --verbose react-native run-ios  
npm verb cli /opt/homebrew/Cellar/node/18.8.0/bin/node /opt/homebrew/lib/node_modules/npm/bin/npm-cli.js
npm info using npm@8.18.0
npm info using node@v18.8.0
npm timing npm:load:whichnode Completed in 0ms
npm timing config:load:defaults Completed in 1ms
npm timing config:load:file:/opt/homebrew/lib/node_modules/npm/npmrc Completed in 2ms
npm timing config:load:builtin Completed in 2ms
npm timing config:load:cli Completed in 2ms
npm timing config:load:env Completed in 0ms
npm timing config:load:file:/Users/steph/Dev/BlueWallet/.npmrc Completed in 1ms
npm timing config:load:project Completed in 4ms
npm timing config:load:file:/Users/steph/.npmrc Completed in 0ms
npm timing config:load:user Completed in 0ms
npm timing config:load:file:/opt/homebrew/etc/npmrc Completed in 0ms
npm timing config:load:global Completed in 0ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:credentials Completed in 1ms
npm timing config:load:setEnvs Completed in 0ms
npm timing config:load Completed in 10ms
npm timing npm:load:configload Completed in 10ms
npm timing npm:load:mkdirpcache Completed in 0ms
npm timing npm:load:mkdirplogs Completed in 1ms
npm verb title npm exec react-native run-ios
npm verb argv "exec" "--loglevel" "verbose" "--" "react-native" "run-ios"
npm timing npm:load:setTitle Completed in 2ms
npm timing config:load:flatten Completed in 1ms
npm timing npm:load:display Completed in 3ms
npm verb logfile logs-max:10 dir:/Users/steph/.npm/_logs
npm verb logfile /Users/steph/.npm/_logs/2022-09-03T06_20_06_276Z-debug-0.log
npm timing npm:load:logFile Completed in 3ms
npm timing npm:load:timers Completed in 0ms
npm timing npm:load:configScope Completed in 0ms
npm timing npm:load Completed in 20ms
npm timing arborist:ctor Completed in 0ms
npm http fetch GET 200 https://registry.npmjs.org/react-native 893ms (cache revalidated)
npm timing arborist:ctor Completed in 0ms
npm timing arborist:ctor Completed in 2ms
npm timing arborist:ctor Completed in 0ms
error Failed to load configuration of your project.
info Run CLI with --verbose flag for more details.
npm timing command:exec Completed in 2255ms
npm verb exit 1
npm timing npm Completed in 2364ms
npm verb code 1
`

`steph@StephMBA Dev % npx --verbose react-native init TestProject
npm verb cli /opt/homebrew/Cellar/node/18.8.0/bin/node /opt/homebrew/lib/node_modules/npm/bin/npm-cli.js
npm info using npm@8.18.0
npm info using node@v18.8.0
npm timing npm:load:whichnode Completed in 0ms
npm timing config:load:defaults Completed in 1ms
npm timing config:load:file:/opt/homebrew/lib/node_modules/npm/npmrc Completed in 1ms
npm timing config:load:builtin Completed in 1ms
npm timing config:load:cli Completed in 2ms
npm timing config:load:env Completed in 0ms
npm timing config:load:file:/Users/steph/Dev/.npmrc Completed in 0ms
npm timing config:load:project Completed in 1ms
npm timing config:load:file:/Users/steph/.npmrc Completed in 0ms
npm timing config:load:user Completed in 0ms
npm timing config:load:file:/opt/homebrew/etc/npmrc Completed in 0ms
npm timing config:load:global Completed in 0ms
npm timing config:load:validate Completed in 0ms
npm timing config:load:credentials Completed in 0ms
npm timing config:load:setEnvs Completed in 1ms
npm timing config:load Completed in 7ms
npm timing npm:load:configload Completed in 7ms
npm timing npm:load:mkdirpcache Completed in 0ms
npm timing npm:load:mkdirplogs Completed in 1ms
npm verb title npm exec react-native init TestProject
npm verb argv "exec" "--loglevel" "verbose" "--" "react-native" "init" "TestProject"
npm timing npm:load:setTitle Completed in 2ms
npm timing config:load:flatten Completed in 1ms
npm timing npm:load:display Completed in 3ms
npm verb logfile logs-max:10 dir:/Users/steph/.npm/_logs
npm verb logfile /Users/steph/.npm/_logs/2022-09-03T06_35_36_449Z-debug-0.log
npm timing npm:load:logFile Completed in 3ms
npm timing npm:load:timers Completed in 0ms
npm timing npm:load:configScope Completed in 0ms
npm timing npm:load Completed in 16ms
npm timing arborist:ctor Completed in 1ms
npm http fetch GET 200 https://registry.npmjs.org/react-native 891ms (cache revalidated)
npm timing arborist:ctor Completed in 0ms
npm timing arborist:ctor Completed in 0ms
npm timing arborist:ctor Completed in 0ms
error Failed to load configuration of your project.
info Run CLI with --verbose flag for more details.
npm timing command:exec Completed in 1784ms
npm verb exit 1
npm timing npm Completed in 1849ms
npm verb code 1
`

Any idea ? What could we look into to help find the problem ?

**Comment by mjvestal** [comment]
This is happening for me when trying to run `npx react-native --verbose` in a folder that is within another project that has a `package.json` (a monorepo).
I noticed that I **could** init a new project in my `Documents` directory just fine, so then I went back to the monorepo folder (a folder up from where I originally tried & failed to init a new React Native project) and deleted the `package-lock.json` and `package.json` files.  This time I was able to successfully `npx react-native init`

**Comment by piyawong** [comment]
i got this bug when i have expo-cli in my global list  try :
yarn global remove expo
yarn global remove expo-cli
npm uninstall -g expo
npm uninstall -g expo-cli

**Comment by rjatin007** [comment]
@piyawong it doesn't work. I even removed expo and expo-cli.

<img width="786" alt="Screenshot 2022-09-24 at 12 13 34 AM" src="https://user-images.githubusercontent.com/30732342/192035858-002fa471-5cc8-4f97-baf3-53972bae7201.png">

**Comment by rjatin007** [comment]
@thymikee Can you please help here?

**Comment by thymikee** [comment]
Do you have a package.json somewhere in Desktop or "rn project" folder? If so, can you remove it? 
Have you tried running the command in a folder name without whitrspaces?

**Comment by rjatin007** [comment]
@thymikee I tried but still no success. Also, no package.json there. not even react-native-cli or react-native-community/cli 
<img width="1680" alt="Screenshot 2022-09-28 at 1 26 52 PM" src="https://user-images.githubusercontent.com/30732342/192722134-bf327f56-2e30-4fb3-9449-6c5d6129de82.png">
<img width="1680" alt="Screenshot 2022-09-28 at 1 27 14 PM" src="https://user-images.githubusercontent.com/30732342/192722219-f7d2a53e-6462-465e-9fc3-5d7e0dbc949d.png">

**Comment by rjatin007** [comment]
@thymikee Any other information you would like me to provide?

**Comment by thymikee** [comment]
Unfortunately I don't have the capacity to help you directly. Maybe there's some kind of antivirus preventing you from doing things? You can try `npx @react-native-community/cli init` and see if that works. Hope someone from the community can step in

**Comment by rjatin007** [comment]
I tried `npx @react-native-community/cli init` but still got same error.
<img width="938" alt="Screenshot 2022-09-29 at 12 52 40 AM" src="https://user-images.githubusercontent.com/30732342/192870496-0cbd5e08-b78b-45fc-b6c1-958efdad7ef6.png">

**Comment by rjatin007** [comment]
`npx create-react-app` is working fine
<img width="650" alt="Screenshot 2022-09-29 at 12 57 42 AM" src="https://user-images.githubusercontent.com/30732342/192871430-3611e055-c59c-4acf-a537-27c4de5edc9a.png">

**Comment by rjatin007** [comment]
@thymikee then issue seems to be in react native cli. Thanks for your time. Can you please tag someone who can help me.

**Comment by thymikee** [comment]
Another thing you could try to pin-point the issue is to install the `@react-native-community/cli` package as a dependency in a different directory, then modify the package's "index" file somewhat like this:
```diff
+     console.log(error);
      throw new CLIError(
        'Failed to load configuration of your project.',
        error,
      );
    }
```
Then call the `path/to/cli/packages/cli/build/bin.js init` and see if the error is logged. Then you should have a better understanding of what's going on. Also, I'd be interested in why the error didn't surface through the CLIError

**Comment by zeejaydev** [comment]
I still get the same error with react-native@0.70.2 cli
as a temp fix you can use an older cli it worked fine for me & you will still get the latest react-naive version 0.70.2
first clean up 
if you use yarn
yarn global remove react-native-cli
yarn global remove react-native
if npm
npm uninstall -g react-native-cli
npm uninstall -g react-native
then
npx react-native@0.68.1 init MyApp

**Comment by adamTrz** [comment]
Hi. New version of cli have been released, could you please give it a go?

**Comment by bfischer1121** [comment]
If you search for "Failed to load configuration of your project." in node_modules, then modify it to:
```ts
throw new (_cliTools().CLIError)(`Failed to load configuration of your project. ${error.message}`, error);
```
it'll show the actual error. It'd be nice to make these changes [here](https://github.com/react-native-community/cli/blob/be3cb6a9fce4eaecfa62179221edf7e4201c89cc/packages/cli/src/index.ts#L174) or [here](https://github.com/react-native-community/cli/blob/be3cb6a9fce4eaecfa62179221edf7e4201c89cc/packages/cli-tools/src/errors.ts). Maybe it's configurable by environment, but for me CLIError is just swallowing the actual error.

**Comment by rjatin007** [comment]
@adamTrz just tried this now, still same error.

<img width="1670" alt="Screenshot 2022-11-01 at 1 27 08 AM" src="https://user-images.githubusercontent.com/30732342/199098977-5263fef8-3c2d-4476-8ce1-11ad0f633a82.png">

**Comment by rjatin007** [comment]
<img width="1236" alt="Screenshot 2022-11-01 at 1 50 52 AM" src="https://user-images.githubusercontent.com/30732342/199102855-7cbafc26-4faf-49b8-988e-7984b0133858.png">

@thymikee  I made directory rn_project then initialised it with npm inti -y. Next I installed the `@react-native-community/cli` package. And  in same directory from VScode terminal I called this command that you wrote.
Apparently error was int my IOS Setup but it did not surface on the CLI (refer screen shot attached)

**Comment by rjatin007** [comment]
@thymikee  Thanks a lot for suggestion, but I think it would have been much easier for me as a developer if it would have surfaced on earlier.

@bfischer1121 Thanks and I agree there should be some displaye of error like this console statement in the index js file too.

Closing this issue with this comment.

**Comment by thymikee** [comment]
@rjatin007 can you create a new feature request or even better a PR with the changes? Error should surface and we need to know why it didn't

**Comment by rjatin007** [comment]
> @rjatin007 can you create a new feature request or even better a PR with the changes? Error should surface and we need to know why it didn't 

@thymikee Sure

**Comment by rjatin007** [comment]
<img width="1674" alt="Screenshot 2022-11-04 at 12 23 13 PM" src="https://user-images.githubusercontent.com/30732342/199910136-10
<img width="1680" alt="Screenshot 2022-11-04 at 12 18 45 PM" src="https://user-images.githubusercontent.com/30732342/199910167-dbd0fca8-70e0-48d4-96c9-5fdafe35783d.png">
12bbbd-7371-47c5-95d8-88c47538ce8c.png">


@thymikee  Sorry for delay in sharing , but i found that even changing the ruby version to 2.7.5 , which was the issue that surfaced, even then the cli doesn't work if its not local one.

In following screenshots first, there is an empty folder named **rn_project** . I have logged ruby version which is 2.7.5.
Next I tried to create project using `npx reat-native init abc` and it should me the old error "faild to loead configuration of your project".

Next I initialised the **rn_project** with `npm init -y` and and installed react-native cli `npm i --save @react-native-community/cli`. then I tried with this cli and it worked.

Now how can I get to the error if local  @react-native-community/cli is working but not npx.

Also `npx react-native run-ios` is giving error
